ra_aid/agent_utils.py

```
@@ -1,113 +1,269 @@
+            max_input_tokens = None
+            logger.debug(f"Could not find token limit for {provider}/{model_name}")
+
+        return max_input_tokens
+
+    except Exception as e:
+        logger.warning(f"Failed to get model token limit: {e}")
+        return None
+
+
+def build_agent_kwargs(
+    checkpointer: Optional[Any] = None,
```

Ruff (0.8.2)

Dynamically typed expressions (typing.Any) are disallowed in `checkpointer`

(ANN401)